types/address/hash.go

```diff
@@ -91,3 +95,121 @@ func Module(moduleName string, derivationKeys ...[]byte) []byte {
 func Derive(address, key []byte) []byte {
 	return Hash(conv.UnsafeBytesToStr(address), key)
 }
+
+// HeimdallHash represents heimdall address
+type HeimdallHash common.Hash
+
+// ZeroHeimdallHash represents zero address
+var ZeroHeimdallHash = HeimdallHash{}
+
+// EthHash get eth hash
+func (aa HeimdallHash) EthHash() common.Hash {
+	return common.Hash(aa)
+}
+
+// Equals returns boolean for whether two HeimdallHash are Equal
+func (aa HeimdallHash) Equals(aa2 HeimdallHash) bool {
+	if aa.Empty() && aa2.Empty() {
+		return true
+	}
+
+	return bytes.Equal(aa.Bytes(), aa2.Bytes())
+}
+
+// Empty returns boolean for whether an AccAddress is empty
+func (aa HeimdallHash) Empty() bool {
+	return bytes.Equal(aa.Bytes(), ZeroHeimdallHash.Bytes())
+}
+
+// Marshal returns the raw address bytes. It is needed for protobuf
+// compatibility.
```

Comment:
> It is needed for protobuf compatibility

Could you elaborate on that need?

Reply:
Still open point. I think this was just copied when heimdall was implemented tbh

Reply:
I see, I'd propose to leave it and, once all modules are forward-ported, we should review which methods aren't used anywhere